### PR TITLE
RCPS-240: Refactor rc-ums

### PR DESCRIPTION
**Pull request by JustineDaveMagnaye**

### CODE REVIEW

File: src/main/java/Main.java.bak [hunk 1]
```diff
@@ -19,7 +16,7 @@
 /**
  * This class represents the main class for the login and registration system.
  */
-public class Main {
+public class Mainjavabak {
```
Comment (RoCS2024) [inline src/main/java/Main.java.bak:19]: why change the class name? just rename the file itself

File: build.properties [hunk 2]
```diff
@@ -1,3 +1,4 @@
 version=1.0
 copyright.year=2024
-Author= Kate Ann I. Amulong
\ No newline at end of file
+Author= Kate Ann I. Amulong
+clean.dirs=build
```
Comment (RoCS2024) [inline build.properties:4]: add / after build

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 3]
```diff
@@ -15,88 +16,81 @@
...
+        return user;
     }
 
     @Override
-    public User saveUser(User User)  {
-
-        String insertQuery = "INSERT INTO login (ID, USERNAME, PASSWORD, ENTITY_ID, DATE_CREATED, DATE_MODIFIED) VALUES (?, ?, ?, ?, ?, ?)";
+    public User saveUser(User user) {
 
         try (Connection connection = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:53]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 4]
```diff
@@ -15,88 +16,81 @@
...
     public static final Logger LOGGER = LoggerFactory.getLogger(UserDaoImpl.class);
+
     @Override
-    public User findUserByUsername(String username){
-        String selectByIdQuery = "SELECT * FROM login WHERE username=?";
-        User User = null;
+    public User findUserByUsername(String username) {
+        User user = null;
 
         try (Connection connection = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:28]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 5]
```diff
@@ -15,88 +16,81 @@
...
+        return user;
     }
 
     @Override
-    public long getMaxUserId(){
-
-        String selectMaxIdQuery = "SELECT MAX(ID) AS MAX_ID FROM login";
+    public long getMaxUserId() {
 
         try (Connection connection = ConnectionHelper.getConnection();
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:74]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 6]
```diff
@@ -15,88 +16,81 @@
...
-    public List<User> getAllUsers(){
 
+    @Override
+    public List<User> getAllUsers() {
         List<User> userList = new ArrayList<>();
-        String query = "SELECT * FROM LOGIN";
 
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(query);
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_ALL_USER_STATEMENT);
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:93]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 7]
```diff
@@ -119,10 +113,8 @@ public List<User> getAllUsers(){
     @Override
     public User getUserById(int id) {
 
-        String sql = "SELECT * FROM login WHERE id = ?";
-
         try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+             PreparedStatement stmt = con.prepareStatement(GET_USER_BY_ID_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:117]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 8]
```diff
@@ -148,11 +140,10 @@ public User getUserById(int id) {
 
     @Override
     public boolean updateUser() {
-        String sql = "UPDATE login SET username = ?, password = ?, entity_id = ?, date_modified = ? WHERE id = ?";
         User user = new User();
 
         try (Connection con = ConnectionHelper.getConnection();
-             PreparedStatement stmt = con.prepareStatement(sql)) {
+             PreparedStatement stmt = con.prepareStatement(UPDATE_USER_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:146]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 9]
```diff
@@ -161,20 +152,19 @@ public boolean updateUser() {
...
     }
 
     @Override
     public User getUsername(String username) {
-        String selectByIdQuery = "SELECT * FROM login WHERE username=?";
         User login = null;
 
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(selectByIdQuery)) {
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_USERNAME_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:167]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 10]
```diff
@@ -215,11 +204,11 @@ public User updatePassword(User user) {
         LOGGER.debug("Updating password failed.");
         return user;
     }
+
     @Override
     public String getPasswordByUsername(String username) {
-        String selectPasswordQuery = "SELECT password FROM login WHERE username=?";
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(selectPasswordQuery)) {
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_PASSWORD_BY_USERNAME_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:211]: fix the try-with-resource. remove preparedstatement from try()

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java [hunk 11]
```diff
@@ -239,10 +228,9 @@ public String getPasswordByUsername(String username) {
 
     @Override
     public String forgotPassword(String username, String newPassword) {
-        String updateQuery = "UPDATE login SET password=?, date_modified=? WHERE username=?";
 
         try (Connection connection = ConnectionHelper.getConnection();
-             PreparedStatement preparedStatement = connection.prepareStatement(updateQuery)) {
+             PreparedStatement preparedStatement = connection.prepareStatement(FORGOT_PASSWORD_STATEMENT)) {
```
Comment (RoCS2024) [inline src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java:233]: fix the try-with-resource. remove preparedstatement from try()

**Review by RoCS2024 — CHANGES_REQUESTED** [review]
see COMMENTS

File: src/main/java/Main.java.bak (hunk 1 above)
Comment (JustineDaveMagnaye) [reply src/main/java/Main.java.bak]: I've renamed the class napo sir

File: build.properties (hunk 2 above)
Comment (JustineDaveMagnaye) [reply build.properties]: I've added the /

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 3 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 4 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 5 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 6 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 7 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 8 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 9 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 10 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

File: src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java (hunk 11 above)
Comment (JustineDaveMagnaye) [reply src/main/java/com/user/management/data/user/dao/impl/UserDaoImpl.java]: okay napo sir

**Comment by RoCS2024** [comment]
Builds are failing
image

**Comment by JustineDaveMagnaye** [comment]
> Builds are failing !

I've resolved the issue napo sir

**Review by RoCS2024 — APPROVED** [review]

**Pull request merged**